qlib/rl/order_execution/strategy.py

```diff
@@ -15,22 +15,294 @@
 
 from qlib.backtest import CommonInfrastructure, Order
 from qlib.backtest.decision import BaseTradeDecision, TradeDecisionWithDetails, TradeDecisionWO, TradeRange
+from qlib.backtest.exchange import Exchange
+from qlib.backtest.executor import BaseExecutor
 from qlib.backtest.utils import LevelInfrastructure
-from qlib.constant import ONE_MIN
-from qlib.rl.data.native import load_backtest_data
+from qlib.constant import EPS, ONE_MIN, REG_CN
+from qlib.rl.data.native import IntradayBacktestData, load_backtest_data
 from qlib.rl.interpreter import ActionInterpreter, StateInterpreter
-from qlib.rl.order_execution.state import SAOEState, SAOEStateAdapter
-from qlib.rl.utils.env_wrapper import BaseEnvWrapper
+from qlib.rl.order_execution.state import SAOEMetrics, SAOEState
+from qlib.rl.order_execution.utils import dataframe_append, price_advantage
 from qlib.strategy.base import RLStrategy
 from qlib.utils import init_instance_by_config
+from qlib.utils.index_data import IndexData
+from qlib.utils.time import get_day_min_idx_range
+
+
+def _get_all_timestamps(
+    start: pd.Timestamp,
+    end: pd.Timestamp,
+    granularity: pd.Timedelta = ONE_MIN,
+    include_end: bool = True,
+) -> pd.DatetimeIndex:
+    ret = []
+    while start <= end:
+        ret.append(start)
+        start += granularity
+
+    if ret[-1] > end:
+        ret.pop()
+    if ret[-1] == end and not include_end:
+        ret.pop()
+    return pd.DatetimeIndex(ret)
+
+
+def fill_missing_data(
```

Comment:
Hope https://github.com/microsoft/qlib/pull/1333 can help to remove this function.